crates/dojo/core/src/world/world_contract.cairo

```diff
@@ -788,6 +807,84 @@ pub mod world {
             }
         }
 
+        fn register_library(
+            ref self: ContractState, namespace: ByteArray, class_hash: ClassHash,
+        ) -> ClassHash {
+            let caller = get_caller_address();
+
+            let namespace_hash = bytearray_hash(@namespace);
+
+            let contract = IDeployedResourceLibraryDispatcher { class_hash };
```

Comment:
The problem here is that we're doing a library call, which can replace the class hash of the world.
I think this is what you meant with the issue about permissions.

Library calls are forbidden in the world as they are not safe.

For a library, we should just accept the class hash, since it can only be registered by someone that has access to the namespace, we have a trust assumption here.

In this context, the contract name may be arbitrary and not directly bound to the library (since we can't do a library call), hence passed as argument. Then the selector is computed from the two inputs (namespace + name).